raft.c

```diff
@@ -2343,27 +2345,73 @@ void handleShardGroupAdd(RedisRaftCtx *rr, RaftReq *req)
     RaftReqFree(req);
 }
 
+void handleShardGroupUpdate(RedisRaftCtx *rr, RaftReq *req)
+{
+    /* Must be done on a leader */
+    if (checkRaftState(rr, req) == RR_ERROR ||
+        checkLeader(rr, req, NULL) == RR_ERROR) {
+        RedisModule_ReplyWithError(req->ctx, "failed, please check logs.");
+        goto exit;
+    }
+
+    /* TODO: validate/handle slot range changes:
```

Comment:
it seems something that is expanding scope.  It would seem to be a followup PR.  we already test for normal overlap, this is new logic for handling migrating/importing state correctly.

Reply:
Is it more than just a single function?

Reply:
I'm just unsure how we do this in a coordinated manner for all the slot types (stable/migrating/importing)